packages/gateway-edge-nextjs/src/cal/cal.ts

```diff
@@ -0,0 +1,150 @@
+import { type NextApiRequest, type NextApiResponse } from "next"
+import ical, { type ICalEventData } from "ical-generator"
+import { type Event } from "@dotkomonline/types"
+import { createServerSideHelpers } from "@trpc/react-query/server"
+import { appRouter, createContextInner, transformer } from "@dotkomonline/gateway-trpc"
+import { getServerSession } from "next-auth";
+import { authOptions } from "../../../auth/src/web.app"
+import jwt from 'jsonwebtoken';
+
+const helpers = createServerSideHelpers({
+  router: appRouter,
+  ctx: await createContextInner({
+    auth: null,
+  }),
+  transformer, // optional - adds superjson serialization
+})
+
+function eventUrl(event: Pick<Event, "id">) {
+  // a better to to get/configure the url?
+  return `https://dev.web.online.ntnu.no/events/${event.id}`
+}
+
+// ALL events
+export async function CalendarAll(req: NextApiRequest, res: NextApiResponse) {
+  if (req.method !== "GET") {
```

Comment:
Are methods not handled at the routing-level? You should not need to do this kind of error-handling yourself IMO

Reply:
cant figure out how to do it in nextjs, maybe @junlarsen knows

Reply:
If we switch to using these we should be able to

https://nextjs.org/docs/app/building-your-application/routing/route-handlers but I think this is fine for now.

Reply:
So you just export as `GET` instead of a default export?

Reply:
<img width="402" alt="image" src="https://github.com/dotkom/monoweb/assets/24441708/ae88ff6a-9e50-4982-bf0d-e64b9310cb4f">

Given we already use this in other places, I think that should be an issue of its own